src/bin/pg_autoctl/cli_service.c

```diff
@@ -424,28 +424,33 @@ cli_service_status(int argc, char **argv)
 
 	if (!file_exists(pathnames->pid))
 	{
-		log_info("pg_autoctl pid file \"%s\" does not exists", pathnames->pid);
+		log_debug("pg_autoctl pid file \"%s\" does not exists", pathnames->pid);
 
 		if (pg_setup_is_running(pgSetup))
 		{
 			log_fatal("Postgres is running at \"%s\" with pid %d",
 					  pgSetup->pgdata, pgSetup->pidFile.pid);
-			exit(EXIT_CODE_INTERNAL_ERROR);
 		}
+
+		log_info("pg_autoctl is not running at \"%s\"", pgSetup->pgdata);
+		exit(PG_CTL_STATUS_NOT_RUNNING);
 	}
 
 	/* ok now we have a pidfile for pg_autoctl */
-	if (read_pidfile(pathnames->pid, &pid))
+	if (read_pidfile(pathnames->pid, &pid) && pid > 0)
 	{
 		if (kill(pid, 0) != 0)
```

Comment:
This check is already done in `read_pidfile`, so it can be removed. When removing this I think none of the `pid > 0` added in this PR are needed anymore.

Reply:
I cleaned the pidfile code a little, in the spirit of https://github.com/citusdata/pg_auto_failover/pull/364 (which we can cancel now), and cleaned the caller code too.

Fixes #364.